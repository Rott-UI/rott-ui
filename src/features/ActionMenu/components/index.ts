export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
