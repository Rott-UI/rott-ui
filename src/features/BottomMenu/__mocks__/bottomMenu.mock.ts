import {BottomMenuItemModel} from '../models'

export const bottomMenuListItemMock: BottomMenuItemModel[] = [
  {
    title: 'Menu 1',
    iconName: 'ARROW_DOWN',
    // imageName: 'CARD_BRAND_MAESTRO',
  },
  {
    title: 'Menu 2',
    iconName: 'ARROW_DOWN',
  },
]

export const menuItems = [
  {title: 'MenuItem1', icon: 'icon1'},
  {title: 'MenuItem2', icon: 'icon2'},
]
