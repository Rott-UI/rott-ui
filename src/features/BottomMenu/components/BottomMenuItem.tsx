// React Imports
import {FC} from 'react'

// React Native Imports
import {Linking} from 'react-native'

// Constant Imports
import {COLOURS} from '../../../constants'

// Component Imports
import {Icon} from '../../Icon'
import {Label} from '../../Label'
import {Image} from '../../Image'
import {Pressable} from '../../Pressable'
import {BottomMenuStyles} from '../styles'
import {BottomMenuItemModel} from '../models'

export const BottomMenuItem: FC<BottomMenuItemModel> = ({
  title,
  iconName,
  imageName,
  options,
  onPress,
  url,
  phone,
}) => {
  return (
    <Pressable
      alignItemsCenter
      width={!options?.containerStyle ? 56 : undefined}
      style={options?.containerStyle}
      onPress={async (event) => {
        if (!!url || !!phone) {
          ;(await Linking.canOpenURL(url ? url! : phone!))
            ? Linking.openURL(url ? url : `tel:${phone}`)
            : null
        } else if (onPress) onPress(event)
      }}>
      {iconName && !imageName && (
        <Icon
          width={options?.width ? options.width : 24}
          height={options?.height ? options.height : 24}
          variant='grey-900'
          mode={options?.iconMode}
          strokeWidth={options?.strokeWidth ? options?.strokeWidth : 2}
          marginBottom={8}
          name={iconName}
        />
      )}

      {imageName && !iconName && (
        <Image
          width={options?.width ? options.width : 24}
          height={options?.height ? options.height : 24}
          tintColor={options?.width && options?.height ? undefined : COLOURS.GREY900}
          marginBottom={8}
          resizeMode='contain'
          name={imageName}
        />
      )}

      {title && (
        <Label
          style={BottomMenuStyles().textStyle}
          variant='grey-900'
          fontSize='xs'
          fontWeight='bold'
          numberOfLines={1}>
          {title}
        </Label>
      )}
    </Pressable>
  )
}
