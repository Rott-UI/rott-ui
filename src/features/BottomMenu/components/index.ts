export * from './BottomMenu'
export * from './BottomMenuItem'
