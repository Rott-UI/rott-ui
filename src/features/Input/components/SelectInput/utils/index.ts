export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
