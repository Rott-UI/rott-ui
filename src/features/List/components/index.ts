export * from './List'
export * from './FavoriteSwipeComponent'
