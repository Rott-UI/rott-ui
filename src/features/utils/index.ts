export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
