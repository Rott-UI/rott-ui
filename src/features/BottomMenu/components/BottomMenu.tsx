// React Imports
import {FC} from 'react'

// Component Imports
import {Content} from '../../Content'
import {BottomMenuItem} from './BottomMenuItem'

// Feature Imports
import {getHasDynamicIslandState, getHasNotchState} from '../../../features/app/appSelector'
import {BottomMenuItemModel} from '../models'

// Hook Imports
import {useAppSelector} from '../../../hooks'
import {COLOURS} from '../../../constants'

interface BottomMenuProps {
  menuItems: BottomMenuItemModel[]
}

export const BottomMenu: FC<BottomMenuProps> = ({menuItems}) => {
  const hasDynamicIsland = useAppSelector(getHasDynamicIslandState)
  const hasNotch = useAppSelector(getHasNotchState)

  return (
    <Content
      testID='bottom-menu-container'
      row
      absolute
      bottom={0}
      noPadding
      paddingTop={11}
      size='full'
      borderTopStartRadius={24}
      borderTopEndRadius={24}
      justifyContentSpaceAround
      backgroundColor={COLOURS.SECONDARY}
      height={hasDynamicIsland || hasNotch ? 88 : 64}>
      {menuItems.map((props) => (
        <BottomMenuItem key={`${props?.title ?? 'Main_'}`} {...props} />
      ))}
    </Content>
  )
}
