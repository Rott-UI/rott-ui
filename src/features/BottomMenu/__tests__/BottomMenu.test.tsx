// Component Imports
import {BottomMenu} from '../components'
import {bottomMenuListItemMock, menuItems} from '../__mocks__'

// Util and Lib Imports
import {render} from '../../../utils'

describe('Component -> BottomMenu', () => {
  const bottomMenuTestId = {
    bottomMenuContainerTestId: 'bottom-menu-container',
  }

  it('bottom menu componenti snapshotı ile eşleşmeli', () => {
    const rendered = render(<BottomMenu menuItems={bottomMenuListItemMock} />)

    expect(rendered).toMatchSnapshot()
  })

  it('ekranda görülmeli', () => {
    const {getByTestId} = render(<BottomMenu menuItems={bottomMenuListItemMock} />)
    const bottomMenuContainer = getByTestId('bottom-menu-container')

    expect(bottomMenuContainer).toBeTruthy()
  })

  it('bottom menu componenti ekranda gözükmeli', () => {
    const {bottomMenuContainerTestId} = bottomMenuTestId

    const {getByTestId} = render(<BottomMenu menuItems={menuItems} />)

    const bottomMenu = getByTestId(bottomMenuContainerTestId)

    expect(bottomMenu).toBeOnTheScreen()
  })
})
