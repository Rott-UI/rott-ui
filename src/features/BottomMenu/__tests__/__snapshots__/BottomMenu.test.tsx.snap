// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> BottomMenu bottom menu componenti snapshotı ile eşleşmeli 1`] = `
<View
  absolute={true}
  borderTopEndRadius={24}
  borderTopStartRadius={24}
  bottom={0}
  height={64}
  paddingTop={11}
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "rgba(255, 199, 44, 1)",
      "borderBottomEndRadius": undefined,
      "borderBottomStartRadius": undefined,
      "borderRadius": undefined,
      "borderTopEndRadius": 24,
      "borderTopStartRadius": 24,
      "bottom": 0,
      "flex": undefined,
      "flexDirection": "row",
      "flexGrow": undefined,
      "gap": undefined,
      "height": 123,
      "justifyContent": "space-around",
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": 123,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": 16,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": 21,
      "paddingVertical": undefined,
      "position": "absolute",
      "right": undefined,
      "top": undefined,
      "width": "100%",
      "zIndex": undefined,
    }
  }
  testID="bottom-menu-container"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    alignItemsCenter={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": 108,
        "zIndex": undefined,
      }
    }
    testID="pressable-test-id"
    width={56}
  >
    <View
      marginBottom={8}
      style={
        {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": 16,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <SvgMock
        fill="transparent"
        height={46}
        stroke="rgba(34, 63, 70, 1)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        width={46}
      />
    </View>
    <Text
      numberOfLines={1}
      style={
        {
          "backgroundColor": undefined,
          "bottom": undefined,
          "color": "rgba(34, 63, 70, 1)",
          "flexGrow": undefined,
          "fontFamily": "Markpro-Bold",
          "fontSize": 10,
          "fontWeight": "bold",
          "gap": undefined,
          "height": undefined,
          "left": undefined,
          "letterSpacing": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "textAlign": "center",
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      Menu 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    alignItemsCenter={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": 108,
        "zIndex": undefined,
      }
    }
    testID="pressable-test-id"
    width={56}
  >
    <View
      marginBottom={8}
      style={
        {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": 16,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <SvgMock
        fill="transparent"
        height={46}
        stroke="rgba(34, 63, 70, 1)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        width={46}
      />
    </View>
    <Text
      numberOfLines={1}
      style={
        {
          "backgroundColor": undefined,
          "bottom": undefined,
          "color": "rgba(34, 63, 70, 1)",
          "flexGrow": undefined,
          "fontFamily": "Markpro-Bold",
          "fontSize": 10,
          "fontWeight": "bold",
          "gap": undefined,
          "height": undefined,
          "left": undefined,
          "letterSpacing": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "textAlign": "center",
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      Menu 2
    </Text>
  </View>
</View>
`;
